internal/endorser/endorser.go

```diff
@@ -76,13 +77,13 @@ func loadAndVerifyProvenances(provenanceURIs []string, binaryDigest string) (*am
 		})
 	}
 
-	if err := verifyProvenances(provenances, binaryDigest); err != nil {
+	if err := verifyProvenances(provenances, reference); err != nil {
 		return nil, fmt.Errorf("verification of provenances failed: %v", err)
 	}
 
 	verifiedProvenances := amber.VerifiedProvenanceSet{
 		BinaryName:   provenances[0].GetBinaryName(),
-		BinaryDigest: binaryDigest,
+		BinaryDigest: reference.BinarySHA256Digests[0],
```

Comment:
This does not sound right. If the reference can contain multiple values for `BinarySHA256Digests`, the value for `BinaryDigest` should not come from the reference, but from the provenances. We also need to make sure that if there are multiple provenances, they should all agree on the binary SHA256 digest.

Reply:
Yes. That makes sense. Thanks :)